### PR TITLE
fix: Different day time check mistake of #6144

### LINKED ISSUE #6237: Issue related to PR #6144
This issue is the extension of the ISSUE  #6117  and PR  #6144 

that pr (created by me) used to resolved the starting and ending time of the event being same, but this should only work for events created for same day, not for the events held for many days.

see the screenshot below , i am creating events for multiple days and i am not able to keep the start time and end time same.

Screenshot_2021-01-05 Create an Event Open Event

**Expected behaviour**
The error or warning should only come for events held for single day not for multiple days.

**Desktop (please complete the following information):**
 - OS: ubuntu 20.04
 - Browser : mozilla firefox

### PR DESCRIPTION
**Pull request by Creatoon**
Fixes #6237 

This was the mistake done by me on the PR #6144.

#### Short description of what this resolves:
Previously , it was not able to create a multiple day event with same starting and ending time. [Check the date]

Screenshot_2021-01-05 Create an Event Open Event(1)

But the error should come for the single day event - [Check the date]

Screenshot_2021-01-05 Create an Event Open Event(2)

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/forms/wizard/basic-details-step.js [hunk 1]
```diff
@@ -142,7 +142,8 @@ export default Component.extend(FormMixin, EventWizardMixin, {
       return parseInt($('.ui.form').form('get value', 'ticket_min_order'), 10) <= parseInt($('.ui.form').form('get value', 'ticket_max_order'), 10);
     };
     $.fn.form.settings.rules.checkValidTimeDifference = () => {
-      return moment($('[name=start_time]')[0].value, 'HH:mm').isBefore(moment($('[name=end_time]')[0].value, 'HH:mm'));
+      if ($('[name=start_date]')[0].value === $('[name=end_date]')[0].value) {return moment($('[name=start_time]')[0].value, 'HH:mm').isBefore(moment($('[name=end_time]')[0].value, 'HH:mm'))}
```
Comment (iamareebjamal) [inline app/components/forms/wizard/basic-details-step.js:145]: Use &&

Comment (Creatoon) [reply app/components/forms/wizard/basic-details-step.js:145]: sir how can i use && in this logic $('[name=start_date]')[0].value === $('[name=end_date]')[0].value

Comment (iamareebjamal) [reply app/components/forms/wizard/basic-details-step.js:145]: That's not what I said. There is no reason to have multiple returns here

**Comment by Creatoon** [comment]
@iamareebjamal review this sir

File: app/components/forms/wizard/basic-details-step.js [hunk 2]
```diff
@@ -142,7 +142,7 @@ export default Component.extend(FormMixin, EventWizardMixin, {
       return parseInt($('.ui.form').form('get value', 'ticket_min_order'), 10) <= parseInt($('.ui.form').form('get value', 'ticket_max_order'), 10);
     };
     $.fn.form.settings.rules.checkValidTimeDifference = () => {
-      return moment($('[name=start_time]')[0].value, 'HH:mm').isBefore(moment($('[name=end_time]')[0].value, 'HH:mm'));
+      return $('[name=start_date]')[0].value === $('[name=end_date]')[0].value ? moment($('[name=start_time]')[0].value, 'HH:mm').isBefore(moment($('[name=end_time]')[0].value, 'HH:mm')) : true;
```
Comment (iamareebjamal) [inline app/components/forms/wizard/basic-details-step.js:145]: Use &&

Comment (Creatoon) [reply app/components/forms/wizard/basic-details-step.js:145]: @iamareebjamal sir please explail this once , i am unable to decode what you are saying.

Comment (iamareebjamal) [reply app/components/forms/wizard/basic-details-step.js:145]: Join the 7 PM meeting. What is your email?

Comment (Creatoon) [reply app/components/forms/wizard/basic-details-step.js:145]: @iamareebjamal yasharyan1099@gmail.com,

**Comment by Creatoon** [comment]
@iamareebjamal sir please review,

**Pull request merged**